### PR TITLE
Feat: initial release of AreaChart

### PR DESCRIPTION
**Pull request by vicky-comeau**

### CODE REVIEW

**Comment by franckgaudin** [comment]
There are some differences between the component popover and the Figma

AreaChart-popover

File: packages/AreaChart/src/chart-tooltip.scss [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    font-size: var(--ids-chart-tooltip-score-font-size);
+    font-weight: var(--ids-chart-tooltip-score-font-weight);
+    margin-right: var(--ids-chart-tooltip-score-value-margin-right);
+  }
+
+  &__circle {
+    background: var(--ids-chart-tooltip-circle-background);
+    border-radius: 100%;
+    flex: 0 0 auto;
+    height: 0.8rem;
```
Comment (ofrogon) [inline packages/AreaChart/src/chart-tooltip.scss:80]: Circle size can be a variable, like the `margin-right`

File: packages/AreaChart/src/AreaChart.tsx [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+
+  const tooltipDateFormatter = (date: number): string => {
+    return DateTime.fromMillis(date).toLocaleString(DateTime.DATE_FULL);
+  };
+
+  const getNumberOfTicks = (linePoints: number, maxTicks: number): number => {
+    let bestTick = 0;
+    let bestScore = 0;
+
+    for (let i = maxTicks; i > 2; i -= 1) {
```
[the remark this answers is not in the document]
Comment (vicky-comeau) [reply packages/AreaChart/src/AreaChart.tsx:135]: actually, the eslint does not let me do this... Normally I would have!

**Comment by franckgaudin** [comment]
I reworked the icons for the doc to be in the same spirit as the others. 

areaChart

File: packages/AreaChart/README.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+const nFormatter = (num) => {
+  const lookup = [
+    { value: 1, symbol: '' },
+    { value: 1e3, symbol: 'k' },
+    { value: 1e6, symbol: 'M' },
+    { value: 1e9, symbol: 'B' },
+    { value: 1e12, symbol: 'T' },
+  ];
+  const digits = 2;
+  const rx = /\.0+$|(\.[0-9]*[1-9])0+$/;
```
[the remark this answers is not in the document]
Comment (vicky-comeau) [reply packages/AreaChart/README.md:39]: When I tried it, it seemed to add an extra 0. I'll just leave as is. It's in the stories anyway. People don't need to use it.

**Comment by ofrogon** [comment]
This build error still happen...
```
[!] (plugin rpt2) Error: /home/runner/work/ov-igloo-ui/ov-igloo-ui/packages/AreaChart/src/AreaChart.tsx(191,23): semantic error TS2551: Property 'grey200' does not exist on type '{ "electric-blue-50": string; "electric-blue-100": string; "electric-blue-200": string; "electric-blue-300": string; "electric-blue-400": string; "electric-blue-500": string; "electric-blue-600": string; ... 121 more ...; "z-index-auto": string; }'. Did you mean 'grey-200'?
```

File: packages/AreaChart/README.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+<AreaChart
+  scoreFormatter={nFormatter}
+  dateRange={{
+    start: '2022-10-01T17:40:40.457Z',
+    end: '2022-10-07T17:40:40.457Z',
+  }}
+  dataSet={[
+    {
+      dateTimeStamp: '2022-10-01T00:40:40.457Z',
```
Comment (franckgaudin) [inline packages/AreaChart/README.md:59]: To have the same format as the other `dateTimeStamp`
```suggestion
      dateTimeStamp: '2022-10-01',
```

File: packages/AreaChart/src/AreaChart.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+      secondaryName: 'sent',
+    },
+    {
+      dateTimeStamp: '2022-10-28T17:40:40.457Z',
+      score: 40066,
+      name: 'sent',
+      secondaryName: 'sent',
+    },
+    {
+      dateTimeStamp: '2022-10-30T17:40:40.457Z',
```
Comment (franckgaudin) [inline packages/AreaChart/src/AreaChart.stories.tsx:209]:
```suggestion
      dateTimeStamp: '2022-10-30',
```

File: packages/AreaChart/src/AreaChart.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+      score: 20000,
+      name: 'sent',
+      secondaryName: 'sent',
+    },
+  ],
+  dateRange: {
+    start: '2022-10-01T17:40:40.457Z',
+    end: '2022-10-30T17:40:40.457Z',
```
Comment (franckgaudin) [inline packages/AreaChart/src/AreaChart.stories.tsx:217]:
```suggestion
    end: '2022-10-30',
```

File: packages/AreaChart/src/AreaChart.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+  range: { min: 'auto', max: 'auto' },
+};
+
+export const OneWeek = () => {
+  return (
+    <Section column>
+      <AreaChart
+        scoreFormatter={nFormatter}
+        dateRange={{
+          start: '2022-10-01T17:40:40.457Z',
```
Comment (franckgaudin) [inline packages/AreaChart/src/AreaChart.stories.tsx:229]:
```suggestion
          start: '2022-10-01',
```

File: packages/AreaChart/src/AreaChart.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+          end: '2022-10-07T17:40:40.457Z',
```
Comment (franckgaudin) [inline packages/AreaChart/src/AreaChart.stories.tsx:230]:
```suggestion
          end: '2022-10-07',
```

File: packages/AreaChart/src/AreaChart.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,285 @@
...
+};
+
+export const ScoreMinMaxRange = () => {
+  return (
+    <Section column>
+      <AreaChart
+        range={{ min: 'dataMin', max: 'dataMax' }}
+        scoreFormatter={nFormatter}
+        dateRange={{
+          start: '2022-10-01T17:40:40.457Z',
```
Comment (franckgaudin) [inline packages/AreaChart/src/AreaChart.stories.tsx:245]:
```suggestion
          start: '2022-10-01',
```

File: packages/AreaChart/src/AreaChart.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,285 @@
...
+          end: '2022-10-07T17:40:40.457Z',
```
Comment (franckgaudin) [inline packages/AreaChart/src/AreaChart.stories.tsx:246]:
```suggestion
          end: '2022-10-07',
```

File: packages/AreaChart/src/AreaChart.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,285 @@
...
+};
+
+export const ScoreAutoRange = () => {
+  return (
+    <Section column>
+      <AreaChart
+        range={{ min: 'auto', max: 'auto' }}
+        scoreFormatter={nFormatter}
+        dateRange={{
+          start: '2022-10-01T17:40:40.457Z',
```
Comment (franckgaudin) [inline packages/AreaChart/src/AreaChart.stories.tsx:261]:
```suggestion
          start: '2022-10-01',
```

File: packages/AreaChart/src/AreaChart.stories.tsx [hunk 12]
```diff
@@ -0,0 +1,285 @@
...
+          end: '2022-10-07T17:40:40.457Z',
```
Comment (franckgaudin) [inline packages/AreaChart/src/AreaChart.stories.tsx:262]:
```suggestion
          end: '2022-10-07',
```

File: packages/AreaChart/src/AreaChart.stories.tsx [hunk 13]
```diff
@@ -0,0 +1,285 @@
...
+  );
+};
+
+export const EmptyData = () => {
+  return (
+    <Section column>
+      <AreaChart
+        scoreFormatter={nFormatter}
+        dateRange={{
+          start: '2022-10-01T17:40:40.457Z',
```
Comment (franckgaudin) [inline packages/AreaChart/src/AreaChart.stories.tsx:276]:
```suggestion
          start: '2022-10-01',
```

File: packages/AreaChart/src/AreaChart.stories.tsx [hunk 14]
```diff
@@ -0,0 +1,285 @@
...
+          end: '2022-10-07T17:40:40.457Z',
```
Comment (franckgaudin) [inline packages/AreaChart/src/AreaChart.stories.tsx:277]:
```suggestion
          end: '2022-10-07',
```

File: website/example/AreaChart.demo.js [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+      ? (num / item.value).toFixed(digits).replace(rx, '$1') + item.symbol
+      : '0';
+  };
+
+  return (
+    <div className="example" style={{ display: 'block' }}>
+      <AreaChart
+        scoreFormatter={nFormatter}
+        dateRange={{
+          start: '2022-10-01T17:40:40.457Z',
```
Comment (franckgaudin) [inline website/example/AreaChart.demo.js:31]:
```suggestion
          start: '2022-10-01',
```

File: website/example/AreaChart.demo.js [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+          end: '2022-10-07T17:40:40.457Z',
```
Comment (franckgaudin) [inline website/example/AreaChart.demo.js:32]:
```suggestion
          end: '2022-10-07',
```

File: website/example/AreaChart.demo.js [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+        }}
+        dataSet={[
+          {
+            dateTimeStamp: '2022-10-01T00:40:40.457Z',
```
Comment (franckgaudin) [inline website/example/AreaChart.demo.js:36]:
```suggestion
            dateTimeStamp: '2022-10-01',
```

**Review by franckgaudin — DISMISSED** [review]

**Review by ofrogon — APPROVED** [review]

**Review by ofrogon — APPROVED** [review]

**Pull request merged**